CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java

```diff
@@ -2179,36 +2175,73 @@ private void updateLayerMask(final int attributeId, final GraphElementType eleme
-                    avoidLayerUpdate = false;
+        } else if (elementType == GraphElementType.TRANSACTION && 
+                transactionFilterBitmaskAttrId != Graph.NOT_FOUND && 
+                transactionLayerVisibilityAttributeId != Graph.NOT_FOUND) {
+            if (attributeId != transactionFilterBitmaskAttrId) {
+                avoidLayerUpdate = true;
+                setIntValue(transactionFilterBitmaskAttrId, elementId, bitmask);
+                avoidLayerUpdate = false;
+            }
+            final float existingVisibility = getFloatValue(transactionLayerVisibilityAttributeId, elementId);
+            if ((bitmask & currentVisibleMask) > 0) {
+                if (existingVisibility != 1.0f) {
+                    attributeDescriptions[transactionLayerVisibilityAttributeId].setFloat(elementId, 1.0f);
+                    attributeIndices[transactionLayerVisibilityAttributeId].updateElement(elementId);
+                    attributeModificationCounters[transactionLayerVisibilityAttributeId] += operationMode.getModificationIncrement();
                 }
-                final float existingVisibility = getFloatValue(transactionLayerVisibilityAttributeId, elementId);
-                if ((bitmask & currentVisibleMask) > 0) {
-                    if (existingVisibility != 1.0f) {
-                        attributeDescriptions[transactionLayerVisibilityAttributeId].setFloat(elementId, 1.0f);
-                        attributeIndices[transactionLayerVisibilityAttributeId].updateElement(elementId);
-                        attributeModificationCounters[transactionLayerVisibilityAttributeId] += operationMode.getModificationIncrement();
-                    }
-                } else {
-                    if (existingVisibility != 0.0f) {
-                        attributeDescriptions[transactionLayerVisibilityAttributeId].setFloat(elementId, 0.0f);
-                        attributeIndices[transactionLayerVisibilityAttributeId].updateElement(elementId);
-                        attributeModificationCounters[transactionLayerVisibilityAttributeId] += operationMode.getModificationIncrement();
-                    }
+            } else {
+                if (existingVisibility != 0.0f) {
+                    attributeDescriptions[transactionLayerVisibilityAttributeId].setFloat(elementId, 0.0f);
+                    attributeIndices[transactionLayerVisibilityAttributeId].updateElement(elementId);
+                    attributeModificationCounters[transactionLayerVisibilityAttributeId] += operationMode.getModificationIncrement();
                 }
             }
         }
     }
 
     // update the layer visibility for an element when an attribute changes
     private void updateLayerMask(final int attributeId, final int elementid) {
+        if(transactionLayerVisibilityAttributeId == Graph.NOT_FOUND) {
+            transactionLayerVisibilityAttributeId = this.addAttribute(
```

Comment:
It feels wrong having these defined in a concept and then copied here. Perhaps we need a low-level concept in the graph framework module that classes like `StoreGraph` can access?

Reply:
I'd have no real idea on what that means for implementation - Would this stop this PR going ahead?

Reply:
As discussed today can we get this merged in and create the low-level concept (which is a great idea by the way) as a separate task @cygnus-x-1? If so please approve this one when you get the chance.